services/table-analyzer.js

```diff
@@ -23,26 +22,53 @@ function TableAnalyzer(queryInterface, config) {
       case 'sqlite':
         query = `PRAGMA foreign_key_list('${table}');`;
         break;
+      default:
+        break;
     }
 
     return queryInterface.sequelize
       .query(query, { type: queryInterface.sequelize.QueryTypes.SELECT });
   }
 
-  function getType(type) {
+  function checkIfColumnIsTypeEnum(columnName) {
+    const query = `
+      SELECT i.udt_name
+      FROM pg_catalog.pg_type t
+      JOIN pg_catalog.pg_namespace n ON n.oid = t.typnamespace
+      JOIN pg_catalog.pg_enum e ON t.oid = e.enumtypid
+      JOIN information_schema.columns i ON t.typname = i.udt_name
+      WHERE i.column_name = '${columnName}'
+      GROUP BY i.udt_name;
+    `;
+
+    return queryInterface.sequelize
+      .query(query, { type: queryInterface.sequelize.QueryTypes.SELECT })
+      .then(result => !!result.length);
+  }
+
+  async function getType(columnInfo, columnName) {
+    const { type, special } = columnInfo;
+    const mysqlEnumRegex = /ENUM\((.*)\)/i;
+
     switch (type) {
-      case 'BIT': // MSSQL type
+      case 'BIT': // MSSQL type
       case 'BOOLEAN':
         return 'BOOLEAN';
       case 'CHARACTER VARYING':
       case 'TEXT':
-      case 'NTEXT': // MSSQL type
-      case 'USER-DEFINED':
+      case 'NTEXT': // MSSQL type
       case (type.match(/TEXT.*/i) || {}).input:
       case (type.match(/VARCHAR.*/i) || {}).input:
       case (type.match(/CHAR.*/i) || {}).input:
-      case 'NVARCHAR': // MSSQL type
+      case 'NVARCHAR': // MSSQL type
         return 'STRING';
+      case 'USER-DEFINED': {
+        const isEnum = await checkIfColumnIsTypeEnum(columnName);
+
+        return isEnum ? `ENUM('${special.join('\', \'')}')` : 'STRING';
+      }
+      case (type.match(mysqlEnumRegex) || {}).input:
+        return type;
       case 'UNIQUEIDENTIFIER':
```

Comment:
There is no enum in mssql. A common way the achieve the same result (and it is what sequelize does) is to create a varchar and create a "check" contraint to only accept certain values.
So here we don't care, since it will be seen as a `STRING`